### ISSUE TITLE
pip_state ignores bin_env

### ISSUE THREAD

**Issue by antmd** [saltstack/salt#48845]
### Description of Issue/Question

I am using SaltStack to install both python2 and 3 packages on three hosts. So, I iterate over a list of package names, but use pip2 first, then pip3.

The pip3 install uses bin_env to select /usr/bin/pip3.

However, for all packages that have python2 versions installed, SaltStack will not install the python3 versions, saying 'Package already installed'.

The output from `salt \* pip.version bin_env=/usr/bin/pip3` shows:

```
minion1:
    8.1.2
minion2:
    8.1.2
master:
    9.0.1
```

Which is incorrect, as pip3 is at version 9.0.1 on all hosts. It is clearly showing the pip2 version numbers, even with the bin_env set.

### Setup

Ensure python2 has packages 'lxml' and 'attrs' installed already, but python3 does not, on all hosts.

Add to a state file that will be run using salt '*' state.apply:

```
{% set python_packages=["lxml", "attrs"] %}
{% for pypi_pkg in python_packages %}
pip3.{{pypi_pkg}}:
  pip.installed:
    - name: {{pypi_pkg}}
    - bin_env: '/usr/bin/pip3'
{% endfor %}
```

### Steps to Reproduce Issue

Run:
```
salt '*' state.apply
```

The output from the state.apply will say 'lxml' and 'attrs' are already installed for python3, even though those packages are installed for python2, only.

### Versions Report

Master and all minions on version 2018.3.2:

```
Salt Version:
           Salt: 2018.3.2
 
Dependency Versions:
           cffi: 1.6.0
       cherrypy: Not Installed
       dateutil: 1.5
      docker-py: 1.10.6
          gitdb: 2.0.0
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: 0.24.3
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: 1.2.5
      pycparser: 2.14
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.24.2
         Python: 2.7.5 (default, Nov  6 2016, 00:28:07)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 13.1.0
           RAET: Not Installed
          smmap: 2.0.1
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: centos 7.3.1611 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-514.6.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.3.1611 Core
```

**Comment by NixM0nk3y** [comment]
I also ran into this in a mixed py2/py3 environment.

Believe this is related to https://github.com/saltstack/salt/commit/3e6e524ecace36ed61664a40160634197ac524db

Fixed for me with:
 
```diff
--- /usr/lib/python3/dist-packages/salt/modules/pip.py  2018-06-22 19:28:56.000000000 +0000
+++ pip.py      2018-07-31 20:50:41.975389643 +0000
@@ -138,7 +138,9 @@
         logger.debug('pip: Using pip from currently-running Python')
         return [os.path.normpath(sys.executable), '-m', 'pip']
 
-    python_bin = 'python.exe' if salt.utils.platform.is_windows() else 'python'
+    python_version = 'python3' if 'pip3' in bin_env else 'python'
+
+    python_bin = '{}.exe'.format(python_version) if salt.utils.platform.is_windows() else python_version

     def _search_paths(*basedirs):
         ret = []
 ```

but this probably doesn't take into account the windows + virtualenv support in the original commit.

**Comment by antmd** [comment]
Looking at the diff you included, I could see that bin_env can take an absolute path to a python executable. I changed my salt state file to:

```
{% set python_packages=["lxml", "attrs"] %}
{% for pypi_pkg in python_packages %}
pip3.{{pypi_pkg}}:
  pip.installed:
    - name: {{pypi_pkg}}
    - bin_env: '/usr/bin/python3.6'
{% endfor %}
```

And it now works. Thanks!

I think the SaltStack documentation should be updated to explain this option, though.

**Comment by NixM0nk3y** [comment]
Great stuff.

Looks like this was reported in https://github.com/saltstack/salt/issues/48122 and should be included in the 2018.3.3 release. So hopefully should be able to work around it till then.
